src/src/camp/camp.service.ts

```diff
@@ -0,0 +1,456 @@
+				return response.status(400).json({
+					success: false,
+					message:
+						'Learners do not belong to the corresponding facilitator.',
+				});
+			}
+
+			let create_group_object = {
+				name: "untitled",
+				type: body.type,
+				status: body.status,
+				program_id: body?.program_id || 1,
+				academic_year_id: body?.academic_year_id || 1,
+				created_by: facilitator_id,
+				updated_by: facilitator_id,
+			};
+			let createresponse = await this.hasuraService.q(
+				'groups',
+				{
+					...create_group_object,
+				},
+				[],
+				false,
+				[
+					...this.returnFieldsgroups,
+					'id',
+					'name',
+					'type',
+					'status',
+				],
+			);
+
+			let group_id = createresponse?.groups?.id;
+			if (createresponse?.groups?.id) {
+				let camp_request_json = {
+					group_id: createresponse?.groups?.id,
+					created_by: facilitator_id,
+					updated_by: facilitator_id,
+				};
+
+				createcampResponse = await this.hasuraService.q(
+					'camps',
+					{
+						...camp_request_json,
+					},
+					[],
+					false,
+					[...this.returnFieldscamps, 'group_id', 'id'],
+				);
+			}
+     
+			let camp_id = createcampResponse?.camps?.id;
+
+			if (!createcampResponse?.camps?.group_id) {
+				await this.hasuraService.delete('groups', {
+					id: group_id,
+				});
+				return response.status(500).json({
+					success: false,
+					data: {},
+					message: 'Camp registration failed.',
+				});
+			}
+
+			// Add group user details for owner or faciltator
+
+			let group_user_owner = {
+				group_id: group_id,
+				user_id: facilitator_id,
+				member_type: 'owner',
+				status: 'active',
+				created_by: facilitator_id,
+				updated_by: facilitator_id,
+			};
+
+			creategroupwoner = await this.hasuraService.q(
+				'group_users',
+				{
+					...group_user_owner,
+				},
+				[],
+				false,
+				[...this.returnFieldsgroupUsers, 'group_id', 'id'],
+			);
+
+			if (!creategroupwoner?.group_users?.id) {
+				await this.hasuraService.delete('camps', {
+					id: camp_id,
+				});
+
+				await this.hasuraService.delete('groups', {
+					id: group_id,
+				});
+
+				return response.status(500).json({
+					success: false,
+					data: {},
+					message: 'error occured during creating group user.',
+				});
+			}
+
+			//add learners to the group users
+			learner_id.forEach(async (id) => {
+				let group_user_member = {
+					group_id: group_id,
+					user_id: id,
+					member_type: 'member',
+					status: 'active',
+					created_by: facilitator_id,
+					updated_by: facilitator_id,
+				};
+
+				await this.hasuraService.q(
+					'group_users',
+					{
+						...group_user_member,
+					},
+					[],
+					false,
+					[...this.returnFieldsgroupUsers, 'group_id', 'id'],
+				);
+			});
+
+			const auditData = {
+				userId: facilitator_id,
+				mw_userid: facilitator_id,
+				context: 'camp.id',
+				context_id: camp_id,
+				oldData: {
+					group_id: group_id,
+					status: body?.status,
```

Comment:
`status: body.status` change to `status: "not_registered`

Reply:
resolved